python-package/xgboost/core.py

```diff
@@ -3143,3 +3144,9 @@ def get_split_value_histogram(
                 UserWarning,
             )
         return nph_stacked
+
+    def __dask_tokenize__(self) -> uuid.UUID:
+        # TODO: Implement proper tokenization to avoid unnecessary re-computation in
+        # Dask. However, default tokenzation causes problems after
+        # https://github.com/dask/dask/pull/10883
+        return uuid.uuid4()
```

Comment:
Two consecutive calls to tokenize() will return two different values, and this can cause problems. I would advise caching the output: https://github.com/dask/dask/issues/11179#issuecomment-2168094529

Reply:
Thanks @crusaderky ! The purpose of this PR was to effectively roll back behavior to "match" tokenization before 10883. However, I agree that it makes sense to cache the value.